### ISSUE TITLE
[JENKINS-29876] NPE when triggering downstream job

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#832]
<p>We've some jobs which do some generic tasks. On success they are used to trigger some downstream jobs. This is similar to <a href="https://issues.jenkins.io/browse/JENKINS-25758" title="NPE when triggering downstream job (again)" class="issue-link" data-issue-key="JENKINS-25758"><del>JENKINS-25758</del></a>, but its resolution didn't help. Downgrade to 1.620 fixed the problem.</p>

<p>There are 2 downstream jobs configured. Starting with 1.621 only the first downstream job gets started after successful build, at best.</p>

<p>Instead of triggering all the downstream jobs they fire this NPE:</p>
<blockquote><p>Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered<br/>
Triggering a new build of .....OUR PROJECT NAME...<br/>
FATAL: null<br/>
java.lang.NullPointerException<br/>
	at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:111)<br/>
	at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:84)<br/>
	at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:141)<br/>
	at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:245)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:687)<br/>
	at hudson.model.Build$BuildExecution.cleanUp(Build.java:200)<br/>
	at hudson.model.Run.execute(Run.java:1788)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:98)<br/>
	at hudson.model.Executor.run(Executor.java:381)</p></blockquote>

---
<details><summary><i>Originally reported by <a title="Marek Lemanczyk" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mlemanczyk">mlemanczyk</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-29876" target="_blank">NPE when triggering downstream job</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Kanstantsin Shautsou" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer">integer</a>
<li><b>reported by</b>: mlemanczyk
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-12-14T00:32:23+00:00
<li><b>votes</b>: 11
<li><b>watchers</b>: 18
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=lindaruthmc">lindaruthmc</a>:</i>
<p>Also having the same issue on v1.624:<br/>
Many of the gateway build jobs are not triggering dependency jobs breaking the build chains for the components. This is 100% for some jobs but not all.</p>

<p>Error when this occurs:<br/>
Notifying upstream projects of job completion<br/>
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered<br/>
FATAL: null<br/>
java.lang.NullPointerException<br/>
at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:111)<br/>
at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:84)<br/>
at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:141)<br/>
at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:245)<br/>
at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:687)<br/>
at hudson.model.Build$BuildExecution.cleanUp(Build.java:200)<br/>
at hudson.model.Run.execute(Run.java:1788)<br/>
at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)<br/>
at hudson.model.ResourceController.execute(ResourceController.java:98)<br/>
at hudson.model.Executor.run(Executor.java:381)</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer">integer</a>:</i>
<p>Please try this war <a href="https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/3228/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/3228/</a> to verify fix</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Kanstantsin Shautsou<br/>
Path:<br/>
 core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java<br/>
 test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/45e4e990269325ba1f8bca25959ff8dcaa21f398" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/45e4e990269325ba1f8bca25959ff8dcaa21f398</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-29876&#93;</span> CheckForNull job in ReverseBuildTrigger</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Oleg Nenashev<br/>
Path:<br/>
 core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java<br/>
 test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/339dd4ace545187d04a65bedea44b215fd11ba39" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/339dd4ace545187d04a65bedea44b215fd11ba39</a><br/>
Log:<br/>
  Merge pull request #1842 from KostyaSha/patch-3</p>

<p><span class="error">&#91;FIXED JENKINS-29876&#93;</span> CheckForNull job in ReverseBuildTrigger</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/5e2b15426118...339dd4ace545" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/5e2b15426118...339dd4ace545</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogfood">dogfood</a>:</i>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/4300/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4300</a><br/>
     <span class="error">&#91;FIXED JENKINS-29876&#93;</span> CheckForNull job in ReverseBuildTrigger (Revision 45e4e990269325ba1f8bca25959ff8dcaa21f398)</p>

<p>     Result = SUCCESS<br/>
kanstantsin.sha : <a href="https://github.com/jenkinsci/jenkins/commit/45e4e990269325ba1f8bca25959ff8dcaa21f398" class="external-link" target="_blank" rel="nofollow noopener">45e4e990269325ba1f8bca25959ff8dcaa21f398</a><br/>
Files : </p>
<ul>
	<li>test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java</li>
	<li>core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>This does not affect 1.625.x LTS as the original change by Kostya which surfaced this issue has been reverted. Integrating it again (+ this fix) will result in behavioral changes without one LTS line, so I doubt that should be done. We'll just have to live that there are entirely too many triggers processed in the 1.625.x line.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Resolved towards 1.632.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer">integer</a>:</i>
<p>LTS is not required imho, let's test this fix in master and get during next LTS or some minor LTS update...</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: tfennelly<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/Util.java<br/>
 core/src/main/java/hudson/console/ConsoleLogFilter.java<br/>
 core/src/main/java/hudson/model/Label.java<br/>
 core/src/main/java/hudson/model/Queue.java<br/>
 core/src/main/java/hudson/model/Run.java<br/>
 core/src/main/java/hudson/slaves/SlaveComputer.java<br/>
 core/src/main/java/hudson/tasks/BuildWrappers.java<br/>
 core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java<br/>
 core/src/main/resources/lib/layout/task.jelly<br/>
 core/src/test/java/hudson/FilePathTest.java<br/>
 core/src/test/java/hudson/LauncherTest.java<br/>
 core/src/test/java/hudson/util/IsOverriddenTest.java<br/>
 core/src/test/java/hudson/util/ProcessTreeTest.java<br/>
 test/src/test/groovy/hudson/model/AbstractProjectTest.groovy<br/>
 test/src/test/groovy/jenkins/model/PeepholePermalinkTest.groovy<br/>
 test/src/test/java/hudson/console/ConsoleLogFilterTest.java<br/>
 test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java<br/>
 test/src/test/java/hudson/model/NodeTest.java<br/>
 test/src/test/java/hudson/model/QueueTest.java<br/>
 test/src/test/java/hudson/slaves/DummyCloudImpl.java<br/>
 test/src/test/java/hudson/slaves/JNLPLauncherTest.java<br/>
 test/src/test/java/hudson/tasks/EnvVarsInConfigTasksTest.java<br/>
 test/src/test/java/hudson/tasks/ShellTest.java<br/>
 test/src/test/java/hudson/tools/JDKInstallerTest.java<br/>
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java<br/>
 test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/955eb7274a02e3549493cd2e1e0de785a0893784" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/955eb7274a02e3549493cd2e1e0de785a0893784</a><br/>
Log:<br/>
  Merge branch 'master' into plugin-manager-dependants</p>

<ul>
	<li>master: (58 commits)<br/>
  Changelog: Replace the PR reference by the JIRA issue reference<br/>
  Noting <a href="https://github.com/jenkinsci/jenkins/pull/1818" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/1818</a><br/>
  Noting <a href="https://github.com/jenkinsci/jenkins/pull/1804" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/1804</a><br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30002" title="Improve Util.isOverridden" class="issue-link" data-issue-key="JENKINS-30002"><del>JENKINS-30002</del></a> Improve Util.isOverridden<br/>
  Noting #1842<br/>
  <span class="error">&#91;FIXED JENKINS-30777&#93;</span> this concludes the fix<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30777" title="ConsoleLogFilter isn&#39;t truly global" class="issue-link" data-issue-key="JENKINS-30777">JENKINS-30777</a> also allow slaves to decorate logger<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30777" title="ConsoleLogFilter isn&#39;t truly global" class="issue-link" data-issue-key="JENKINS-30777">JENKINS-30777</a> Generalized the signature to work with Run, not just AbstractBuild.<br/>
  Diamond operator<br/>
  <span class="error">&#91;FIXED JENKINS-29876&#93;</span> CheckForNull job in ReverseBuildTrigger<br/>
  Noting <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> in changelog<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> remove extra space<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> address feedbacks<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> indent back<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> fixing test<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> fixing test<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> enhancing test case<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> test added to make sure a flyweight task can be blocked at last minute<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> fix regression when flyweight task is blocked by upstream/downstream project<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-30084" title="FlyWeightTasks tied to a label will not cause node provisioning and will be blocked forever." class="issue-link" data-issue-key="JENKINS-30084"><del>JENKINS-30084</del></a> some more polish<br/>
  ...</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer">integer</a>:</i>
<p>Your stacktrace ends in Result.java please firstly check and ensure that it the same issues before commenting.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>That looks more like <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a>. Make sure saving the job config (and selecting a threshold) doesn't fix it, then file a new issue and link to <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-30132">JENKINS-30132</a>]</i>
